### PR TITLE
8304412: Serial: Refactor old generation cards update after Full GC

### PR DESCRIPTION
**Pull request by albertnetymk**
Simple refactoring to use more informative names/comments.

Test: tier1-6

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8304412](https://bugs.openjdk.org/browse/JDK-8304412): Serial: Refactor old generation cards update after Full GC

### Reviewers
 * [Thomas Schatzl](https://openjdk.org/census#tschatzl) (@tschatzl - **Reviewer**)
 * [Kim Barrett](https://openjdk.org/census#kbarrett) (@kimbarrett - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/13076/head:pull/13076` \
`$ git checkout pull/13076`

Update a local copy of the PR: \
`$ git checkout pull/13076` \
`$ git pull https://git.openjdk.org/jdk.git pull/13076/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 13076`

View PR using the GUI difftool: \
`$ git pr show -t 13076`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/13076.diff">https://git.openjdk.org/jdk/pull/13076.diff</a>

</details>

### CODE REVIEW

**Review by tschatzl — CHANGES_REQUESTED** [review]

File: src/hotspot/share/gc/serial/genMarkSweep.cpp [hunk 1]
```diff
@@ -109,22 +109,12 @@ void GenMarkSweep::invoke_at_safepoint(bool clear_all_softrefs) {
...
-    rs->clear_into_younger(old_gen);
-  } else {
-    // Invalidate the cards corresponding to the currently used
-    // region and clear those corresponding to the evacuated region.
-    rs->invalidate_or_clear(old_gen);
-  }
+  // Update old gen cards to maintain old-to-young-pointer invariant: Clear
+  // the old generation card table completely if the young generation had been
+  // completely evacuated, otherwise dirties the whole old generation to
+  // conservatively not loose any old-to-young pointer.
```
Comment (tschatzl) [inline src/hotspot/share/gc/serial/genMarkSweep.cpp:115]: I was actually thinking of putting this into the hpp file at the declaration of the method.

**Review by tschatzl — APPROVED** [review]
lgtm. Please consider moving the comment.

**Review by tschatzl — APPROVED** [review]
Thanks!

**Review by kimbarrett — APPROVED** [review]
Looks good.

**Comment by albertnetymk** [comment]
Thanks for the review.

/integrate

**Pull request closed without merging**